src/Compilers/Core/CodeAnalysisTest/AnalyzerAssemblyLoaderTests.cs

```diff
@@ -1677,8 +1675,30 @@ public void AssemblyLoadingInNonDefaultContext_AnalyzerReferencesSystemCollectio
 
                 Assert.Equal("42", sb.ToString());
             });
+        }
 
-            alc.Unload();
+        [Theory]
+        [CombinatorialData]
+        public void AssemblyLoading_DoesNotUseCollectibleALCs(AnalyzerTestKind kind)
+        {
+            // When an assembly is loaded into a collectible AssemblyLoadContext it prevents any of the R2R logic from being used.
```

Comment:
Put a blurb that this R2R is critical to our VS / CLI perfromance scenarios. Basically a place for some future dev to look for context if they ever want to change this.